### PR TITLE
manually check compile args against spec

### PR DESCRIPTION
**Pull request by hiredman**
This change stops lacinia from turnning on spec assert checking globally
and manully checks the compile-args spec when compile is called.

### CODE REVIEW

File: src/com/walmartlabs/lacinia/schema.clj [hunk 1]
```diff
@@ -1210,20 +1214,16 @@
   ([schema]
    (compile schema nil))
   ([schema options]
+   ;; This is based on clojure.spec's assert, but is always on
+   ;; the single branch alt adds :args to the explain path as expected in tests
+   (when-not (s/valid? ::compile-args [schema options])
+     (let [ed (s/explain-data ::compile-args [schema options])]
+       (throw (ex-info
+               (str "Spec assertion failed\n" (with-out-str (s/explain-out ed)))
```
Comment (hlship) [inline src/com/walmartlabs/lacinia/schema.clj]: Also, this looks like a bit of code that might be refactored out to internal-utils for re-use elsewhere.

Comment (hlship) [inline src/com/walmartlabs/lacinia/schema.clj]: How about `Arguments to compile do not conform to spec:`  ?

Comment (hiredman) [reply src/com/walmartlabs/lacinia/schema.clj]: looking at explain-data, it returns nil if the spec is invalid, so we don't need the extra valid? check, which simplifies things. I also added the custom exception message. Those two things together have me leaning towards not pulling it out as a utility, but let me know if you would still prefer that.

**Comment by hlship** [comment]
It is necessary for you to sign the CLA (contributor license agreement) before we can merge.

Technically, you must sign the CLA before we can review the PR, but the CLA requirement is newer than the PR.

Please re-open when you are ready to sign the CLA.

**Comment by hlship** [comment]
Please re-open this PR when you are ready to sign the CLA.  I'd love to merge this.

**Comment by bcarrell** [comment]
Looks like some conflicts after I merged the other PR.  Mind handling?

**Pull request merged**